ttnn/cpp/ttnn/operations/experimental/paged_cache/paged_cache.hpp

```diff
@@ -0,0 +1,50 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "device/paged_cache_operation.hpp" // TODO: not right!
+#include "ttnn/run_operation.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/compute_kernel_config.hpp"
+#include "ttnn/core.hpp"
+
+namespace ttnn {
+namespace operations::experimental::paged_cache {
+
+struct PagedUpdateCacheOperation {
```

Comment:
Can we split these into two?

* ttnn
  * operations
    * experimental
      * paged_cache
        * common (if needed)
        * paged_fill_cache
        * paged_update_cache

Reply:
They have different kernels, programs, device operations.

Reply:
They share the same op device struct, just like `update_cache` and `fill_cache`.

Reply:
Just a suggestion, if you think current structure is more proper, I am good

Reply:
I appreciate the thorough look you're giving the code :) I think the current structure is alright - there's just enough shared code between the two ops that it makes sense to me to have everything but the program factory be shared.